dora/core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -163,6 +189,94 @@ public boolean isSeekable() {
     return false;
   }
 
+  @Nullable
+  @Override
+  public Iterator<UfsStatus> listStatusIterable(
+      String path, ListOptions options, String startAfter, int batchSize) throws IOException {
+    // Calling this method on non s3 UFS might result in OOM because batch based fetching
+    // is not supported and this method essentially fetches all ufs status and converts it to
+    // an iterator.
+    UfsStatus[] result = listStatus(path, options);
+    if (result == null) {
+      return null;
+    }
+    Arrays.sort(result, Comparator.comparing(UfsStatus::getName));
```

Comment:
Is this sort required for non s3 ufs?

Reply:
not necessarily but we can just let the interface always return a sorted result. I can probably update the interface desc later.

Reply:
actually let me remove this

Reply:
actually i removed this